### PR TITLE
@orta => Facebook login/signup

### LINKED ISSUE #2: Support Facebook login / sign up
Currently done via the `FBSession openActiveSessionWithReadPermissions` API in Eigen. 

This could be optimal, but maybe it also can use the social framework?  I didn't write the original implementation so I don't know the constraints that took us to this design.

### PR DESCRIPTION
**Pull request by ashfurrow**
The diff looks a bit mucky. Basically I've broken down the Facebook signin/signup process into atomic, reusable chunks defined in the `Facebook_Private` class extension. There were some other issues with general XApp usage that I fixed, too. 

Fixes #2.

### CODE REVIEW

File: Artsy+Authentication.podspec [hunk 1]
```diff
@@ -17,5 +17,5 @@ Pod::Spec.new do |s|
   s.source_files = 'Pod/Classes'
   s.private_header_files = 'Pod/Classes/*Private.h'
   s.frameworks = 'Foundation', 'Social', 'Accounts'
-  s.dependencies = ['ISO8601DateFormatter', 'NSURL+QueryDictionary', 'TwitterReverseAuth']
+  s.dependencies = ['ISO8601DateFormatter', 'NSURL+QueryDictionary', 'TwitterReverseAuth', 'AFNetworking']
```
Comment (orta) [inline Artsy+Authentication.podspec]: No AFNetworking pleasssssse, lets keep this as unassuming as possible about deps.

Comment (orta) [inline Artsy+Authentication.podspec]: this doesn't have to be in this PR, but I'd rather this lib not make you go down one route for this stuff

Comment (ashfurrow) [inline Artsy+Authentication.podspec]: ... sounds reasonable. Any objection to including a lightweight, private network wrapper?

Comment (orta) [inline Artsy+Authentication.podspec]: NSURLSession is good for tiny things like this ( wouldn't build an app on it though )

**Comment by orta** [comment]
Looks good. Gonna be a badass though and say, not merging on tests.

bwahahahahhahaaha

**Comment by ashfurrow** [comment]
What kind of testing do you think would be appropriate in this case? Stubbing data and verifying error codes/tokens? Anything beyond that?

**Comment by orta** [comment]
The dream, which I don't think we should do right now is to have a running tests against staging that:
- Creates a user via Twitter
- Creates a user via Facebook
- Logs in via Twitter
- Logs in via Facebook
- Creates a user via email + pass
- Logs in using email + pass
- Deletes all those accounts

For now, things like "does the right callbacks", "calls the right paths" and "has the right params" seem to be the things we should cover.

**Comment by ashfurrow** [comment]
:warning: Empty tests for now – do not merge even if :green_apple:

**Comment by ashfurrow** [comment]
@orta So I think this is sufficiently test except for the new Facebook stuff, which is a bit tricky. I already have the `ACAccountStore` as an associated object on `ArtsyAuthentication`, so I think subclassing it should be sufficient to inject some different behaviour in.

**Comment by ashfurrow** [comment]
@orta I've got this tested to the point where I'm happy with it – merge when you feel it's ready, too.

**Comment by orta** [comment]
that testing? fabulous.

**Pull request merged**